lib/Controller/AvatarController.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * @copyright Copyright (c) 2020 Matthias Heinisch <contacts@matthiasheinisch.de>
+ *
+ * @author Matthias Heinisch <contacts@matthiasheinisch.de>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Contacts\Controller;
+
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http\TemplateResponse;
+// use OCP\IInitialStateService;
+use OCP\IConfig;
+use OCP\L10N\IFactory;
+use OCP\IRequest;
+
+class AvatarController extends Controller {
+
+	protected $appName;
+
+	// /** @var IInitialStateService */
+	// private $initialStateService;
+
+	/** @var IFactory */
+	private $languageFactory;
+	/** @var IConfig */
+	private  $config;
+
+	public function __construct(string $AppName,
+								IRequest $request,
+								IConfig $config,
+								// IInitialStateService $initialStateService,
+								IFactory $languageFactory) {
+		parent::__construct($AppName, $request);
+
+		$this->appName = $AppName;
+		// $this->initialStateService = $initialStateService;
+		$this->languageFactory = $languageFactory;
+		$this->config = $config;
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 *
+	 * Overview page to update avatars from social media
+	 * for a complete addressbook
+	 */
+	public function view(): TemplateResponse {
```

Comment:
Let's not do a page for it as mentioned in another commend and create an ApiController file to receive a proper sync social request from the front :)
That would clean things up as well ;)

Other than this, we definitely see you followed the tutorial or experienced the way we work here, your overall syntax is good!

Reply:
Alright